pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -243,34 +244,17 @@ func GenerateEC2NodeClassManifest(ctx context.Context) *awsv1beta1.EC2NodeClass
 // CreateNodeClaim creates a nodeClaim object.
 func CreateNodeClaim(ctx context.Context, nodeClaimObj *karpenterv1.NodeClaim, kubeClient client.Client) error {
 	klog.InfoS("CreateNodeClaim", "nodeClaim", klog.KObj(nodeClaimObj))
-	return retry.OnError(retry.DefaultBackoff, func(err error) bool {
-		return err.Error() != consts.ErrorInstanceTypesUnavailable
-	}, func() error {
-		err := CheckNodeClass(ctx, kubeClient)
-		if err != nil {
-			return err
-		}
-
-		err = kubeClient.Create(ctx, nodeClaimObj.DeepCopy(), &client.CreateOptions{})
-		if err != nil {
-			return err
-		}
-		time.Sleep(1 * time.Second)
-
-		updatedObj := &karpenterv1.NodeClaim{}
-		err = kubeClient.Get(ctx, client.ObjectKey{Name: nodeClaimObj.Name, Namespace: nodeClaimObj.Namespace}, updatedObj, &client.GetOptions{})
+	err := CheckNodeClass(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
 
-		// if SKU is not available, then exit.
-		_, conditionFound := lo.Find(updatedObj.GetConditions(), func(condition status.Condition) bool {
-			return condition.Type == karpenterv1.ConditionTypeLaunched &&
-				condition.Status == metav1.ConditionFalse && strings.Contains(condition.Message, consts.ErrorInstanceTypesUnavailable)
-		})
-		if conditionFound {
-			klog.Error(consts.ErrorInstanceTypesUnavailable, "reconcile will not continue")
-			return fmt.Errorf(consts.ErrorInstanceTypesUnavailable)
-		}
+	err = kubeClient.Create(ctx, nodeClaimObj.DeepCopy(), &client.CreateOptions{})
+	if err != nil {
 		return err
-	})
+	}
+
+	return nil
```

Comment:
how about optimize as following:

```
return kubeClient.Create(ctx, nodeClaimObj, &client.CreateOptions{})
```

I think there is no need to deepcopy object, because Create function will not modify object.

Reply:
You are right, fixed.